nextjs-13/src/components/AppLayout/index.tsx

```diff
@@ -1,10 +1,48 @@
+'use client';
+
+import classNames from 'classnames';
+import Link from 'next/link';
+import { redirect, usePathname } from 'next/navigation';
+import { useSession } from 'next-auth/react';
+
 import Header from '@components/Header';
 
 export default function Layout({ children }: { children: React.ReactNode }) {
+  const pathname = usePathname();
+  const isRoot = !pathname.includes('/send');
+  const { status } = useSession();
+
+  if (status === 'unauthenticated') {
+    redirect('/auth/sign-in');
+  }
```

Comment:
I noticed that this code was removed above https://github.com/nimblehq/js-framework-benchmark/pull/123/files#diff-e8b3129d5740bf43e4514d1c0d8e01bf9e8906c58721b29fb66dba4891d140d1. While this can be managed at the layout level, what do you think of using [the middleware](https://github.com/nimblehq/js-framework-benchmark/blob/develop/nextjs-13/src/middleware.ts) for that purpose? It would clean up component files.

Reply:
At first, when I read the [NextAuth middleware page](https://next-auth.js.org/configuration/nextjs#middleware), I missed this [wrap middleware part](https://next-auth.js.org/configuration/nextjs#wrap-middleware), so I thought NextAuth only allow us to use middleware in a very [basic way](https://next-auth.js.org/configuration/nextjs#basic-usage) 👀 Let me try this wrap middleware way!

Reply:
It turned out that that wrap middleware is not ideal also 🤣, so I decided to just use the [normal next.js middleware way](https://github.com/nimblehq/js-framework-benchmark/pull/123/commits/0e71f45a42ac09949e190ab90a347df4d9994e2f#diff-8dc469b9cf60785ce08e724cab36052f5f6278de35da8aea179287ddbe5c6ac3R4-R19)

Reply:
Next.js middleware's capabilities are very limited indeed (there can be only one 🤯 ), but it usually suffices for simple applications. The improvement looks good.